alchemy/src/sqlite/sqlite-state-store.ts

```diff
@@ -0,0 +1,158 @@
+import type { Config as LibSQLConfig } from "@libsql/client";
+import type { Options as BetterSQLite3Options } from "better-sqlite3";
+import type { Database } from "bun:sqlite";
+import fs from "node:fs";
+import path from "node:path";
+import type { Scope } from "../scope.ts";
+import { memoize } from "../util/memoize.ts";
+import {
+  BaseSQLiteStateStore,
+  resolveMigrationsPath,
+} from "./internal/base-sqlite-state-store.ts";
+import * as schema from "./internal/schema.ts";
+
+type BunSQLiteOptions = ConstructorParameters<typeof Database>[1];
+
+interface BunSQLiteStateStoreOptions {
+  engine: "bun";
+  filename?: string;
+  options?: BunSQLiteOptions;
+}
+
+interface BetterSQLite3StateStoreOptions {
+  engine: "better-sqlite3";
+  filename?: string;
+  options?: BetterSQLite3Options;
```

Comment:
We can flatten the `BetterSQLite3Options` into the interface

Reply:
Done